### PR TITLE
Deduplicate scan summaries

### PR DESCRIPTION
**Pull request by oheger-bosch**
This PR prevents the creation of duplicate entities for scan summaries in cases where a single scan operation yields multiple results.

### CODE REVIEW

File: workers/scanner/src/test/kotlin/OrtServerScanResultStorageTest.kt [hunk 1]
```diff
@@ -101,7 +73,7 @@ class OrtServerScanResultStorageTest : WordSpec() {
 
                 verifyAssociatedScanResults(scannerRun, scanResult)
 
-                scanResultStorage.read(provenance, scannerMatcher) shouldBe listOf(scanResult)
+                scanResultStorage.read(provenance.mapToOrt(), scannerMatcher) shouldBe listOf(scanResult)
```
Comment (nnobelis) [inline workers/scanner/src/test/kotlin/OrtServerScanResultStorageTest.kt:84]: Why these changes with `mapToOrt` ? They seem unrelated to this commit.

File: workers/scanner/src/main/kotlin/scanner/ScanSummaryHash.kt [hunk 2]
```diff
@@ -0,0 +1,78 @@
+/*
```
Comment (nnobelis) [inline workers/scanner/src/main/kotlin/scanner/ScanSummaryHash.kt:1]: Why create a full featured hash function instead of an `equals` one ?

You don't store the hash anywhere and you'd save on memory because you would not need to build a huge string.

File: workers/scanner/src/test/kotlin/OrtServerScanResultStorageTest.kt (hunk 1 above)
Comment (oheger-bosch) [reply workers/scanner/src/test/kotlin/OrtServerScanResultStorageTest.kt:84]: `mapToOrt` is now needed, since the data types have changed. The tests have to deal with both ORT types and ORT Server types. Test data is now created using the model of ORT Server, but the scan result storage expects types from ORT's model.

File: workers/scanner/src/main/kotlin/scanner/ScanSummaryHash.kt (hunk 2 above)
Comment (oheger-bosch) [reply workers/scanner/src/main/kotlin/scanner/ScanSummaryHash.kt:1]: The hashes were used because again different types (ORT model and ORT Server model) are involved. But you are right, this is not the most efficient approach. Will try to come up with a better solution.

Comment (oheger-bosch) [reply workers/scanner/src/main/kotlin/scanner/ScanSummaryHash.kt]: I have added a comparison function that should be more efficient.

**Review by nnobelis — APPROVED** [review]
